src/api/File.js

```diff
@@ -40,30 +43,45 @@ class File extends Item {
     }
 
     /**
-     * API for getting download URL for files
+     * API for getting download URL for files and file versions
      *
-     * @param {string} id - File id
+     * @param {string} fileId - File id
+     * @param {BoxItem|BoxItemVersion} fileOrFileVersion - File or file version to download
      * @param {Function} successCallback - Success callback
      * @param {Function} errorCallback - Error callback
      * @return {void}
      */
-    getDownloadUrl(id: string, successCallback: Function, errorCallback: ElementsErrorCallback): Promise<void> {
+    async getDownloadUrl(
+        fileId: string,
+        fileOrFileVersion: BoxItem | BoxItemVersion,
+        successCallback: string => void,
+        errorCallback: ElementsErrorCallback,
+    ): Promise<void> {
         this.errorCode = ERROR_CODE_GET_DOWNLOAD_URL;
-        this.successCallback = successCallback;
         this.errorCallback = errorCallback;
-        return this.xhr
-            .get({
-                url: this.getUrl(id),
-                params: {
-                    fields: FIELD_DOWNLOAD_URL,
-                },
-            })
-            .then(({ data }: { data: BoxItem }) => {
-                this.successHandler(data[FIELD_DOWNLOAD_URL]);
-            })
-            .catch((e: $AxiosError<any>) => {
-                this.errorHandler(e);
-            });
+        this.successCallback = successCallback;
+
+        const downloadAvailable = fileOrFileVersion[FIELD_IS_DOWNLOAD_AVAILABLE];
+        const downloadUrl = fileOrFileVersion[FIELD_AUTHENTICATED_DOWNLOAD_URL];
+
+        if (!downloadAvailable || !downloadUrl) {
+            this.errorHandler({ code: this.errorCode, message: 'Item to download missing required fields.' });
+            return;
+        }
+
+        const token: TokenLiteral = await TokenService.getReadToken(getTypedFileId(fileId), this.options.token);
+        const tokenString: ?string = token && (typeof token === 'string' ? token : token.read);
```

Comment:
Is this check needed? It looks like `getReadToken` already does a check to see if token is an object and to return the read property

Reply:
Indeed, nice catch. I was overcomplicating the return value due to a blind copy-paste. Will fix in the next commit.